### PR TITLE
refactor: make importing templates available as a lib

### PR DESCRIPTION
**Pull request by luisdavim**
## Description

These changes are to allow using the template importing functionality available as a library.
I didn't add or removed any functionality, only moved things around.

### CODE REVIEW

**Review by dougm — APPROVED** [review]
Thanks @luisdavim , refactor looks great. Didn't realize how much helper code we'd accumulated over the years for this functionality!

Regarding the license headers, I don't know why it's complaining about files you haven't touched. The checker does expect a range even if same year (e.g. `2024-2024`)
You can also run this script locally, which only adjusts files changed in your branch.
https://github.com/vmware/govmomi/blob/main/scripts/license.sh

File: go.mod [hunk 1]
```diff
@@ -11,6 +11,7 @@ require (
 	github.com/stretchr/testify v1.9.0
 	github.com/vmware/vmw-guestinfo v0.0.0-20170707015358-25eff159a728
 	github.com/xlab/treeprint v1.2.0
+	golang.org/x/text v0.16.0
```
[the remark this answers is not in the document]
Comment (luisdavim) [reply go.mod:14]: yes, it's the Title case, the function that was being used before has been deprecated for a while now.

**Comment by luisdavim** [comment]
I've ran `./scripts/license.sh` and pushed the resulting changes.

**Review by dougm — APPROVED** [review]
Thanks @luisdavim

**Comment by slimm609** [comment]
@dougm can you please merge when possible?

**Pull request merged**